### DISCUSSION TITLE
How to debug "No matching expectation found"?

### DISCUSSION THREAD

**Discussion by nitnelave** [asomers/mockall#386] [Questions] [answered]
First of all, great crate, thanks a lot!

When I write a new test, if I don't get the expectations exactly right, I get this error. It would be useful to have a printout of the arguments passed that didn't have any matching expectations, whenever possible, otherwise I'm stuck reading through the code and modifying things until it passes.

**Comment by asomers** [comment, marked as answer]
That would be very useful.  But it isn't possible, at least not with stable Rust.  But if you build Mockall with the **nightly** feature, _and_ your arguments types implement `Debug`, then you'll get what you want.
